src/codeu/chat/ServerMain.java

```diff
@@ -96,9 +96,12 @@ private static void runServer(Uuid id,
                         new RemoteRelay(relaySource);
 
     final Server server = new Server(id, secret, relay);
-
     LOG.info("Created server.");
 
+    // before running the server, we must first read from the logger.
+    server.retrieveState();
+    LOG.info("Last saved state has been retrieved.");
```

Comment:
Since retrieveState will run on the main executing thread while the actual logging will run in another one, this LOG will return before the work has been done.